### PR TITLE
Use published apps in IIS tests

### PR DESCRIPTION
**Pull request by pakrym**
Fixes: https://github.com/aspnet/AspNetCore/issues/7025
Fixes: https://github.com/aspnet/AspNetCore/issues/3950

### CODE REVIEW

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/PublishedApplicationPublisher.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Server.IntegrationTesting;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests
+{
+    public class PublishedApplicationPublisher: ApplicationPublisher
```
Comment (jkotalik) [inline src/Servers/IIS/IIS/test/Common.FunctionalTests/PublishedApplicationPublisher.cs]: Don't like the name. But I don't have an alternative.

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/PublishedApplicationPublisher.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    {
+        private readonly string _applicationPath;
+
+        public PublishedApplicationPublisher(string applicationPath):base(applicationPath)
```
Comment (jkotalik) [inline src/Servers/IIS/IIS/test/Common.FunctionalTests/PublishedApplicationPublisher.cs]: nit spacing on base

**Review by jkotalik — APPROVED** [review]
💰

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/Inprocess/StartupTests.cs [hunk 3]
```diff
@@ -183,18 +183,15 @@ public async Task HelloWorld(TestVariant variant)
...
         {
             var deploymentParameters = _fixture.GetBaseDeploymentParameters(_fixture.InProcessTestSite, publish: true);
             // We need the right dotnet on the path in IIS
             deploymentParameters.EnvironmentVariables["PATH"] = Path.GetDirectoryName(DotNetCommands.GetDotNetExecutable(deploymentParameters.RuntimeArchitecture));
-
-            // rest publisher as it doesn't support additional parameters
-            deploymentParameters.ApplicationPublisher = null;
             // ReferenceTestTasks is workaround for https://github.com/dotnet/sdk/issues/2482
-            deploymentParameters.AdditionalPublishParameters = "-p:RuntimeIdentifier=win7-x64 -p:UseAppHost=true -p:SelfContained=false -p:ReferenceTestTasks=false";
+            deploymentParameters.AdditionalPublishParameters = "AppHost";
```
Comment (natemcmaster) [inline src/Servers/IIS/IIS/test/Common.FunctionalTests/Inprocess/StartupTests.cs]: Seems like we're misusing what `AdditionalPublishParameters` is meant for (msbuild args). I think this deserves a new property: `DeploymentParameters.PublishProfile`

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/PublishedSitesFixture.cs [hunk 4]
```diff
@@ -19,13 +19,11 @@ public class PublishedSitesCollection : ICollectionFixture<PublishedSitesFixture
...
     public class PublishedSitesFixture : IDisposable
     {
-        public CachingApplicationPublisher InProcessTestSite { get; } = new CachingApplicationPublisher(Helpers.GetInProcessTestSitesPath());
-        public CachingApplicationPublisher OutOfProcessTestSite { get; } = new CachingApplicationPublisher(Helpers.GetOutOfProcessTestSitesPath());
+        public PublishedApplicationPublisher InProcessTestSite { get; } = new PublishedApplicationPublisher(Helpers.GetInProcessTestSitesPath());
+        public PublishedApplicationPublisher OutOfProcessTestSite { get; } = new PublishedApplicationPublisher(Helpers.GetOutOfProcessTestSitesPath());
 
         public void Dispose()
         {
-            InProcessTestSite.Dispose();
```
Comment (natemcmaster) [inline src/Servers/IIS/IIS/test/Common.FunctionalTests/PublishedSitesFixture.cs]: Why is this no longer needed?

Comment (pakrym) [reply src/Servers/IIS/IIS/test/Common.FunctionalTests/PublishedSitesFixture.cs]: We were publishing and catching published output before. This was a way to clean up the cache which is no longer required.

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/Inprocess/StartupTests.cs (hunk 3 above)
Comment (pakrym) [reply src/Servers/IIS/IIS/test/Common.FunctionalTests/Inprocess/StartupTests.cs]: It's a bit of layering thing. I don't want to add DeploymentParameters property that would be only used by this specific implementation in IIS.

**Comment by pakrym** [comment]
@muratg test infrastructure work limited to IIS test assets

**Pull request merged**